wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -48,14 +51,21 @@ getFullWalletHistory cWalId = do
                 cWalId
             pure mempty
 
-    localHistory <- getLocalHistory addrs
+    let localHistory =
+            DL.fromList $ filterLocalTh
+                (DL.toList blockHistory)
+                (DL.toList unfilteredLocalHistory)
 
     logTxHistory "Block" blockHistory
     logTxHistory "Mempool" localHistory
 
     fullHistory <- addRecentPtxHistory cWalId $ DL.toList $ localHistory <> blockHistory
     cHistory <- forM fullHistory $ addHistoryTx cWalId
     pure (cHistory, fromIntegral $ length cHistory)
+  where
+    filterLocalTh blockH localH = do
+        let blockTxIds = map _thTxId blockH
+        filter ((`notElem` blockTxIds) . _thTxId) localH
```

Comment:
This should use a `Set` and `Set.notMember`. Also, a type signature on `filterLocalTh` would clarify which kind of list we're using (ie normal list not `DList`).

Note also that this is a simple pure function, it does not need `do`, that's confusing. Just use `let ... in ...`

Reply:
Thanks, fixed.